Model/Order/PlaceOrder.php

```diff
@@ -247,10 +247,14 @@ public function authorizeAndPlace(string $publicOrderId, string $quoteMaskId): R
         $websiteId = (int)$this->storeManager->getStore()->getWebsiteId();
         $shopId = $this->config->getShopId($websiteId) ?? '';
 
-        try {
-            $quoteId = $this->maskedQuoteIdToQuoteId->execute($quoteMaskId);
-        } catch (NoSuchEntityException $exception) {
-            return $this->getValidationErrorResponse((string)__('Invalid quote mask ID "%1"', $quoteMaskId));
+        if (!is_numeric($quoteMaskId) && strlen($quoteMaskId) === 32) {
```

Comment:
^So not sure how this would work? 
Are you asking the sdk to pass in the quoteID from the database because that feels like a security risk? 
Converting the quoteID and adding a quote mask ID seems counter to what Magento does, unless I'm missing something?

Reply:
As long as there is an active user session, their currently active cart is loaded using their customer ID. This code allows a real quote ID to be used in place of a masked ID. Nothing needs to be passed from the SDK to the endpoint to make this work. 

Magento's built-in API endpoints also use multiple routes to call the same service method in this manner. As Caitlin pointed out, the value passed to the service method is set by the configured Parameter Overrider object (I'm reusing the [ParamOverriderCartId](https://github.com/magento/magento2/blob/2.4.7-p1/app/code/Magento/Quote/Model/Webapi/ParamOverriderCartId.php) object from the core). The only difference is that we need to be able to accept one of two parameters based on which route is called, so we have to determine which parameter we received in the service method (via the conditional statement that I added).

I've verified that this works using HTTPie by requesting a customer authentication token and then using that token to authenticate against the new endpoint. As long as that authentication token is active somewhere on the frontend (it would have to be for us to call other "me" and "mine" endpoints belonging to Magento), this will suit our purposes.

Reply:
^So for the frontend APIs we can't actually grab the customer token since we need the user login and password to retrieve it. Magento doesn't use this either and has a work-around for this that we tap into here: 
https://github.com/bold-commerce/checkout-experience-templates-internal/blob/cdc99fabedf0c80d0f2214a647f5d384f6fb5fdd/templates/src/themes/flow-sdk/magento/getMagentoUrl.ts#L16 

I think we'd still need to update the resource 
            <resource ref="Magento_Sales::create"/>
^This is my understanding of how the the other flows get around this.

Reply:
In your linked code, you're calling the `/V1/carts/mine` API endpoints, which all use the [ParamOverriderCartId](https://github.com/magento/magento2/blob/2.4.7-p1/app/code/Magento/Quote/Model/Webapi/ParamOverriderCartId.php) object to override the passed cart ID with the customer's cart ID. They all use [routes](https://github.com/magento/magento2/blob/2.4.7-p1/app/code/Magento/Quote/etc/webapi.xml#L44-L79) with the same configuration as mine to achieve that. My API endpoint should therefore work the same way as Magento's endpoints do.

Reply:
As the only difference between the two use request is how we retrieve the quote, this method is acceptable. If there was a dramatic change in the underlying logic for the two operations I would agree to split them out.